tests/dla/test_lie_closure.py

```diff
@@ -268,3 +271,143 @@ def test_is_independent(self, ops, op, is_independent_true, tol):
         assert v1._pw_to_idx == vcopy._pw_to_idx
         assert v1._rank == vcopy._rank
         assert v1._num_pw == vcopy._num_pw
+
+
+dla11 = [
+    PauliSentence({PauliWord({0: "X", 1: "X"}): 1.0, PauliWord({0: "Y", 1: "Y"}): 1.0}),
+    PauliSentence(
+        {
+            PauliWord({0: "Z"}): 1.0,
+        }
+    ),
+    PauliSentence(
+        {
+            PauliWord({1: "Z"}): 1.0,
+        }
+    ),
+    PauliSentence({PauliWord({0: "Y", 1: "X"}): -1.0, PauliWord({0: "X", 1: "Y"}): 1.0}),
+]
+
+
+class TestLieClosure:
+    """Tests for qml.dla.lie_closure()"""
+
+    def test_verbose(self, capsys):
+        """Test the verbose output"""
+        gen11 = dla11[:-1]
+        _ = lie_closure(gen11, verbose=True)
+        captured = capsys.readouterr()
+        assert captured.out == "epoch 1 of lie_closure\nepoch 2 of lie_closure\n"
+
+    def test_simple_lie_closure(self):
+        """Test simple lie_closure example"""
+        gen11 = dla11[:-1]
+        res11 = lie_closure(gen11, pauli=True)
+        assert res11 == dla11
+
+        dla12 = [
+            PauliSentence({PauliWord({0: "X", 1: "X"}): 1.0, PauliWord({0: "Y", 1: "Y"}): 1.0}),
+            PauliSentence(
+                {
+                    PauliWord({0: "Z"}): 1.0,
+                }
+            ),
+            PauliSentence({PauliWord({0: "Y", 1: "X"}): -1.0, PauliWord({0: "X", 1: "Y"}): 1.0}),
+            PauliSentence({PauliWord({0: "Z"}): -2.0, PauliWord({1: "Z"}): 2.0}),
+        ]
+        gen12 = dla12[:-1]
+        res12 = lie_closure(gen12, pauli=True)
+        assert PauliVSpace(res12) == PauliVSpace(dla12)
```

Comment:
Why  do we need the VSpace here?

Reply:
In that case indeed `res12 == dla12` is also `True`, but I want to test that they span the same subspace, not that they yield the exact same operators (which may change due to future changes)

Reply:
Fair enough :)